Iceberg Integration documentation

MeelahMe opened the pull request:
**Description**

To introduce documentation for the new Iceberg integration with Snowflake for InfluxDB Cloud Dedicated. To create documentation that allows users to export time series data snapshots into Apache Iceberg format and query them directly from Snowflake—without the need for complex ETL processes.

**To-do**

- [x] Feature overview and benefits
- [x] Create steps to request and configure the integration via InfluxData Sales and Support
- [x] Explain details on how the InfluxData team sets up the Snowflake external stage and catalog
- [x] Share a export process overview, including current limitations (e.g., customer-triggered exports not yet supported)
- [x] Explain guidance for querying Iceberg tables in Snowflake
- [x] Talk about considerations and known limitations

**Why This Matters**

This documentation gives users the information they need to request access and understand the setup and usage of the integration clearly and confidently.

MeelahMe committed:
Iceberg Integration

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+## Overview
+
+Snapshots for Snowflake enable users to export time-series data from InfluxDB into a structured format using Apache Iceberg. This integration facilitates efficient data sharing between InfluxDB and Snowflake without the need for complex ETL processes.
```
```suggestion
Export time-series data snapshots from InfluxDB into Apache Iceberg format.
Integrate data with Snowflake and other Iceberg-compatible tools without the need for complex ETL processes.
```

- "Snapshots for Snowflake" suggests that this is the feature name.
Let's keep Snowflake prominent, but primarily emphasize exporting in Iceberg format (which most of the page does).
- Use second person
- Avoid big words like "facilitates"

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+- Access to an **external object store** (such as AWS S3).
+- Familiarity with **Apache Iceberg** and **Snowflake**.
+
+## Step-by-step guide
```
```suggestion
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+## Step-by-step guide
+
+### Step 1: Configure external storage
```
```suggestion
## Step 1: Configure external storage
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+For more details, refer to the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/tables-iceberg-configure-catalog-integration-object-storage).
+
+### Step 2: Set up a catalog integration in Snowflake
```
```suggestion
## Step 2: Set up a catalog integration in Snowflake
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+Set up a catalog integration in Snowflake to manage and load Iceberg tables efficiently.
+
+#### Example: Create a catalog integration in Snowflake
```
```suggestion
### Example: Create a catalog integration in Snowflake
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+Set up an external storage location (such as AWS S3) to store Iceberg table data and metadata.
+
+#### Example: Configure an S3 stage in Snowflake
```
```suggestion
### Example: Configure an S3 stage in Snowflake
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
````diff
@@ -0,0 +1,152 @@
+  ENABLED = TRUE;
+```
+
+For more details, refer to the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/tables-iceberg-configure-catalog-integration).
````
```suggestion
For more information, refer to the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/tables-iceberg-configure-catalog-integration).
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+For more details, refer to the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/tables-iceberg-configure-catalog-integration).
+
+### Step 3: Export InfluxDB data to Iceberg format
```
```suggestion
## Step 3: Export InfluxDB data to Iceberg format
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+### Step 3: Export InfluxDB data to Iceberg format
+
+Use InfluxDB's Iceberg Exporter to convert and export your time-series data into the Iceberg table format.
```
```suggestion
Use the InfluxData's Iceberg exporter to convert and export your time-series data from your {{% product-name omit="Clustered" %}} cluster into the Iceberg table format.
```

Use the specific version name (`product-name` shortcode); the exporter isn't compatible with all versions (I assume). This becomes even more important for relying on AI to "deduce" feature support.

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+Use InfluxDB's Iceberg Exporter to convert and export your time-series data into the Iceberg table format.
+
+#### Example: Export data using the Iceberg Exporter
```
```suggestion
### Example: Export data using the Iceberg exporter
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
````diff
@@ -0,0 +1,152 @@
+cd influxdb_iox/iceberg_exporter
+```
+
+- Configure the exporter with your InfluxDB data source and target Iceberg table location.
````
Add an example here and description of "InfluxDB data source"

(Important, but shouldn't prevent you releasing this)

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+- Configure the exporter with your InfluxDB data source and target Iceberg table location.
+- Run the exporter to generate Iceberg-compatible Parquet files.
+
+For more details, refer to the [Iceberg Exporter README](https://github.com/influxdata/influxdb_iox/tree/main/iceberg_exporter).
```
Add an example for running the exporter.

(Important, but shouldn't prevent you releasing this)

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
````diff
@@ -0,0 +1,152 @@
+  METADATA_FILE_PATH = 's3://my-bucket/path/to/metadata.json';
+```
+
+Ensure that `EXTERNAL_VOLUME` and `METADATA_FILE_PATH` point to your external storage and metadata file.
````
I would describe these in separate bullet points.

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+Ensure that `EXTERNAL_VOLUME` and `METADATA_FILE_PATH` point to your external storage and metadata file.
+
+### Step 5: Query the Iceberg table from Snowflake
```
```suggestion
## Step 5: Query the Iceberg table from Snowflake
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+Once the Iceberg table is set up, you can query it using standard SQL in Snowflake.
+
+#### Example: Query the Iceberg table
```
```suggestion
### Example: Query the Iceberg table
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+## Interfaces for using Iceberg integration
+
+- **CLI**: `Influx CTL` enables users to trigger snapshot exports.
+- **API**: Provides REST endpoints to manage and configure snapshots.
+- **SQL (Snowflake)**: Query Iceberg tables using standard SQL.
```
```suggestion
- [Use the CLI to trigger snapshot exports](#use-the-CLI-to-trigger-snapshot-exports)
- [Use the API to manage and configure snapshots](#use-the-api-to-manage-and-configure-snapshots)
- [Use SQL in Snowflake to query Iceberg tables](#use-sql-in-snowflake-to-query-iceberg-tables)
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+- **API**: Provides REST endpoints to manage and configure snapshots.
+- **SQL (Snowflake)**: Query Iceberg tables using standard SQL.
+
+## CLI and API reference
+
```
```suggestion
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+
+## CLI and API reference
+
+### CLI commands
```
```suggestion
### Use the CLI to trigger snapshot exports
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
````diff
@@ -0,0 +1,152 @@
+influxctl export --namespace foo --table bar
+```
+
+### API endpoints
````
```suggestion
### Use the API to manage and configure snapshots

Use the {{% product-name %}} HTTP API to export snapshots and check status.
```

We try to be specific about _HTTP API_ (vs Flight query APIs)

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,152 @@
+- **Performance**: Query performance may vary based on data size and query complexity.
+- **Feature support**: Some advanced features of InfluxDB may not be fully supported in Snowflake through Iceberg integration.
+
+## Next steps
+
+- Enhance REST Catalog support.
+- Optimize AWS S3 access control and security.
+- Improve Grafana dashboards and alerting.
+- Expand compatibility testing with other Iceberg engines.
+
+## References
+
+- **InfluxDB Iceberg Exporter**: [GitHub Repository](https://github.com/influxdata/influxdb_iox/tree/main/iceberg_exporter)
+- **Snowflake Iceberg Tables**: [Snowflake Documentation](https://docs.snowflake.com/en/user-guide/tables-iceberg)
```
```suggestion
```

We don't want to forecast features or changes and you already have the links in context.

jstirnaman reviewed (changes requested):
Overall, this looks good. My suggestions are mostly for following our style and "tightening up". We should add examples where noted ASAP, but it shouldn't necessarily hold up publishing when the feature is ready.

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe commented:
Thank you for the suggestions and help! I should probably go slower and triple check that I am using the Google developers documentation style guide. @jstirnaman

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

jstirnaman commented:
> Thank you for the suggestions and help! I should probably go slower and triple check that I am using the Google developers documentation style guide. @jstirnaman

Ideally, Vale should be catching violations as well, but I think we're missing some config for the newer directories.
You can tell CoPilot to always use it and suggest fixes. I also give it an alias, like

> If I say "gdd", it means Google Developer Documentation style.

```
Use gdd and condense the selection...
```

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,140 @@
+
+### Key Benefits
+
+- **Efficient data access**: Query InfluxDB data directly from Snowflake.
```
```suggestion
- **Efficient data access**: Query your data directly from Snowflake.
```

We avoid saying "InfluxDB data", because it's the customer's data.

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,140 @@
+- **Efficient data access**: Query InfluxDB data directly from Snowflake.
+- **Cost-effective storage**: Optimize data retention and minimize storage costs.
+- **Supports AI and ML workloads**: Enhance machine learning applications by making time-series data accessible in Snowflake.
+
```
We should remove the "Step <num>" from the headings and add a numbered-list TOC for the steps here.

jstirnaman commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,140 @@
+
+## Step 3: Export InfluxDB data to Iceberg format
+
+Use the InfluxData's Iceberg exporter to convert and export your time-series data from your {{% product-name omit="Clustered" %}} cluster into the Iceberg table format.
```
```suggestion
Use InfluxData's Iceberg exporter to convert and export your time-series data from your {{% product-name omit="Clustered" %}} cluster into the Iceberg table format.
```

jstirnaman reviewed (changes requested):
A few more minor suggestions

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Jason Stirnaman <jstirnaman@influxdata.com>

MeelahMe committed:
updating headings

MeelahMe committed:
Adding a numbered-list TOC

MeelahMe committed:
Added an export command example

MeelahMe committed:
Proof read: improving grammar and clarity

MeelahMe committed:
Adding explanations for examples

MeelahMe committed:
Updating  examples

sanderson commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,198 @@
+
+For more information, refer to the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/tables-iceberg-configure-catalog-integration).
+
+## Export InfluxDB data to Iceberg format
```
Side note, this is where Dedicated and Clustered are going to diverge, so we'll likely need unique docs for each. I believe we will be doing the data export for Dedicated customers, but Clustered users will have to do it for themselves.

sanderson commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
````diff
@@ -0,0 +1,198 @@
+
+### Use the CLI to trigger snapshot exports
+
+#### Example: Enable Iceberg feature and export a snapshot
+
+```sh
+# Enable Iceberg feature
+$ influxctl enable-iceberg
+
+# Export a snapshot
+$ influxctl export --namespace foo --table bar
+```
````
This commands should be used in the process above, not `influxdb_iox`

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

MeelahMe committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

ritwika314 commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,198 @@
+This example assumes the following:
+
+- You have followed the example for [writing and querying data in the IOx README](https://github.com/influxdata/influxdb_iox/blob/main/README.md#write-and-read-data).
+- You've configured compaction to trigger more quickly with these environment variables:
+  - `INFLUXDB_IOX_COMPACTION_MIN_NUM_L0_FILES_TO_COMPACT=1`
+  - `INFLUXDB_IOX_COMPACTION_MIN_NUM_L1_FILES_TO_COMPACT=1`
```
These are just internal testing instructions, lets not include these.

ritwika314 commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
````diff
@@ -0,0 +1,198 @@
+
+#### Running the export command
+
+```console
+$ influxdb_iox iceberg export \
+  --catalog-dsn postgresql://postgres@localhost:5432/postgres \
+  --source-object-store file 
+  --source-data-dir ~/.influxdb_iox/object_store \
+  --sink-object-store file \
+  --sink-data-dir /tmp/iceberg \
+  --export-config-path config.json
+```
````
No these are local testing instructions . i am going through this PR now, sorry i was buried under a lot of notifications and had not noticed this. I will provide detailed instructions by EOD.

ritwika314 reviewed (changes requested):
Added a few comments, we should save the material for when more self-support features are added. Thank you!

ritwika314 commented on content/shared/influxdb3-query-guides/snapshots/snowflake.md:
```diff
@@ -0,0 +1,176 @@
+
+### Create a Snowflake external stage
+
+Use the `CREATE STAGE` Snowflake SQL command to set up an external storage location
+(such as AWS S3) to store Iceberg table data and metadata--for example:
```
Yes, thats along the lines of what i was thinking. The flow would be pretty much the same, Contact sales to get iceberg.  Once your tables have been set up for iceberg, some details on how your support engineers will set up export and help set up the catalog integrations in snowflake...

MeelahMe committed:
removing run export example and adding

MeelahMe committed:
Changes to- Example: Export data using Iceberg exporter

MeelahMe committed:
Adding note to- Export InfluxDB time series data to Iceberg format

MeelahMe committed:
removing reduandant sections and information

MeelahMe committed:
removing IOX references

MeelahMe committed:
Rewriting section to focus and clarify that Iceberg table is done with support assistance

MeelahMe committed:
removing entier section: Interfaces for using Iceberg integration section

MeelahMe committed:
Adding read-only note to section

MeelahMe committed:
updating TOC numbe4r list

MeelahMe committed:
Updating who to contact for Snowflake Integration

MeelahMe committed:
Restructuring, reving, and proof reading edit

MeelahMe committed:
google dev style read through

MeelahMe committed:
updating prereqs

MeelahMe committed:
wip: removing tab shortcode to figure out error

MeelahMe committed:
adding shortcode structure back

jstirnaman committed:
feat(dedicated): iceberg export for snowflake:
- Add directory and frontmatter
- Revise content to be Cloud Dedicated specific, omitting implementation and config detail that will be handled by InfluxData Support
- Fix code samples
- Organize, clean up headings

- TODO: Comment out influxctl and API code if necessary (until features are completed)
- TODO: Verify details with PM
- TODO: Move content as noted in the iceberg-for-snowflake.md comment

jstirnaman reviewed (commented):
In addition to the notes here, I'm pushing an update to your branch--I'll PR so you can see the diff.

jstirnaman committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

jstirnaman committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

jstirnaman committed:
Update content/shared/influxdb3-query-guides/snapshots/snowflake.md

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

jstirnaman committed:
Apply suggestions from code review

Co-authored-by: Scott Anderson <sanderson@users.noreply.github.com>

jstirnaman committed:
feat(dedicated): iceberg for snowflake:
- Merge fixes
- Provide instruction for checking snapshot status
- Cleanup
